### PR TITLE
add apcu enabled check if apcu extension loaded (#10310)

### PR DESCRIPTION
**Pull request by aleksejs1**

### CODE REVIEW

**Comment by derrabus** [comment]
Can you take a look at the CI failures?

**Comment by aleksejs1** [comment]
I have no idea why is CI fails... apcu enabled in setup, but looks like apcu_enabled() return false.
If ini_get will return the same, then need some more time for investigation

File: lib/Doctrine/ORM/ORMSetup.php [hunk 1]
```diff
@@ -181,7 +182,7 @@ private static function createCacheInstance(
 
         $namespace = 'dc2_' . md5($proxyDir);
 
-        if (extension_loaded('apcu')) {
+        if (extension_loaded('apcu') && ini_get('apc.enabled')) {
```
Comment (greg0ire) [inline lib/Doctrine/ORM/ORMSetup.php]:
```suggestion
        if (extension_loaded('apcu') && apcu_enabled()) {
```

File: lib/Doctrine/ORM/Tools/Setup.php [hunk 2]
```diff
@@ -236,7 +237,7 @@ private static function createCacheInstance(bool $isDevMode, ?Cache $cache): Cac
 
         if ($isDevMode === true) {
             $cache = class_exists(ArrayCache::class) ? new ArrayCache() : new ArrayAdapter();
-        } elseif (extension_loaded('apcu')) {
+        } elseif (extension_loaded('apcu') && ini_get('apc.enabled')) {
```
Comment (greg0ire) [inline lib/Doctrine/ORM/Tools/Setup.php]:
```suggestion
        } elseif (extension_loaded('apcu') && apcu_enabled()) {
```

**Comment by greg0ire** [comment]
It makes little sense to enable APCu in the CLI. Because of that, [there is another setting](https://www.php.net/manual/en/apcu.configuration.php#ini.apcu.enable-cli) if you want to do it anyway.

I think you should set that setting in the failing tests, because it does not make much sense to use `AcpuAdapter` if that cache system is not actually available with the CLI.

**Comment by aleksejs1** [comment]
> I think you should set that setting in the failing tests

Sorry, but how to enable apcu-cli in failing tests? Should it done in ci-configs, or in test annotation?

**Comment by greg0ire** [comment]
That setting is `PHP_INI_SYSTEM`, so I'm not sure it can be done in test annotations. In fact, I must confess I didn't know this was doable with test annotations, can you link to the relevant PHPUnit docs?

Anyway, I think I would use https://github.com/shivammathur/setup-php#ini-values-optional in this file: https://github.com/doctrine/orm/blob/2.13.x/.github/workflows/continuous-integration.yml

File: .github/workflows/continuous-integration.yml [hunk 3]
```diff
@@ -66,7 +66,7 @@ jobs:
           php-version: "${{ matrix.php-version }}"
           extensions: "apcu, pdo, ${{ matrix.extension }}"
           coverage: "pcov"
-          ini-values: "zend.assertions=1"
+          ini-values: "zend.assertions=1, apc.enable_cli=1"
```
Comment (greg0ire) [inline .github/workflows/continuous-integration.yml:69]: Look for more occurrences of `zend.assertions=1` in the file to see where you might need to add this.

**Review by derrabus — APPROVED** [review]

**Pull request merged**